easybuild/tools/module_generator.py

```diff
@@ -325,6 +362,21 @@ def set_alias(self, key, value):
         # quotes are needed, to ensure smooth working of EBDEVEL* modulefiles
         return 'set-alias\t%s\t\t%s\n' % (key, quote_str(value))
 
+    def lmod_only(self, statements):
+        """Put provided statements under Lmod-only condition."""
+        # $_ contains full path to program being used to parse module file
+        # when using Lmod, this should always be the tcl2lua.tcl script
+        return self.conditional_statement('string match "*tcl2lua.tcl" $env(_)', statements, indent_body=False)
```

Comment:
quoting @rtmclay:

```
I like the trick with using $_ to figure out whether Lmod or Tmod is parsing a TCL modulefile.   This trick works on Linux and Mac OS X and Solaris.  It would probably be good to check that all the various version of Unix you are trying to run Lmod also support $_.

HOWEVER, the internals of Lmod have changed.  The program tcl2lua.tcl is no longer called directly.  Instead it is /path/to/tclsh /path/to/tcl2lua.tcl  is called.  So I suggested that you use this instead:

     if { [string match "*modulecmd*" $env(_)]} {
       #Tmod
    } else {
       #Lmod
    }

The string *modulecmd* will match both modulecmd and modulecmd.tcl.  This if block will work with older and newer version of Lmod.
```

(cfr. https://sourceforge.net/p/lmod/mailman/message/35396987/)